ChainSafe/web3.unity
# Marketplace fixes

Transaction values weren't being converted to hex integers properly. Closes #998
Fixed marketplace TX hashes not returning (they were returning internally but the user may want them exposed). Closes #997

Also updated EVMs send transaction and reworked it's input + tests to match the above as that'd be incorrect also, cleaned up the TODO there, tests are passing in the editor.

To Test:

Boot up the sample scene
Test marketplace calls

Expected Results:

Tx hashes are displayed with marketplace write calls
NFTs can be purchased from the marketplace

Note:
Be aware that the marketplace's ID to purchase is the NFTs index ID in the marketplace's list, not the actual NFT ID. So if you see NFT ids as 1, 0, 0 from left to right in your marketplace, their corresponding index ids are 0, 1 and 2. The index ID is what needs to be entered unity side in the serialized field itemID.

![image](https://github.com/ChainSafe/web3.unity/assets/57473220/6a709e61-a2bc-4b06-9895-327825c067ef)

Talking to Arun to see if we can get the dash updated to display the item ID index as well as it's NFT ID to avoid user confusion.
